src/remap/function/replace.rs

```diff
@@ -108,6 +108,7 @@ impl Expression for ReplaceFn {
 
                 Ok(replaced)
             }
+            Argument::Array(_) => unreachable!(),
```

Comment:
It'd be nice to put a small message alongside these unreachables (eg "function does not accept this arg type" for future readers.

Reply:
I changed how arrays are handled in https://github.com/timberio/vector/pull/5308, so this will be removed once that PR is merged.